src/reason_pprint_ast.ml

```diff
@@ -2597,9 +2597,21 @@ class printer  ()= object(self:'self)
         | (_::_, None) -> add_bar nameOf (normalize args)
         | (_::_, Some res) -> add_bar nameOf (normalize (args@[res]))
       in
+      (*
+        Attach attributes sitting on the entire variant leaf.
+          Example:
+            [ | `utf8 | `my_name [@bs.as "ascii"]]
+            type variantType = | Foo int [@onInt]
+        We don't want [@bs.as "ascii"], nor [@onInt] dropped. Because of [@bs.as "ascii"]
+        coming in from the Ast nodes outside of the Parsetree.constructor_declaration, while
```

Comment:
You're right, I misinterpreted the whole thing. Thanks for pointing this out! Is the overall approach ok here?